packages/workflow/src/internals.ts

```diff
@@ -478,8 +478,12 @@ export class Activator implements ActivationHandler {
     // Most things related to initialization have already been handled in the constructor
     this.mutateWorkflowInfo((info) => ({
       ...info,
-      searchAttributes:
-        (mapFromPayloads(searchAttributePayloadConverter, searchAttributes?.indexedFields) as SearchAttributes) ?? {},
+
+      // TODO(thomas): do we not filter out undefined values?
```

Comment:
Not sure what you meant exactly by that comment, but current impl looks ok to me. On `initializeWorkflow`, it safe to assume that the server wouldn't send us "undefined" SA values.

Reply:
```
export function decodeSearchAttributes(indexedFields: Record<string, Payload> | undefined | null): SearchAttributes {
  if (!indexedFields) return {};
  return Object.fromEntries(
    Object.entries(mapFromPayloads(searchAttributePayloadConverter, indexedFields) as SearchAttributes).filter(
      ([_, v]) => v && v.length > 0
    ) // Filter out empty arrays returned by pre 1.18 servers
  );
}
```
When we `decodeSearchAttributes` from server (i.e. in `executionInfoFromRaw`) we make sure to filter out undefined values and empty arrays. I assumed that server could send undefined SA values because of this, or is this some legacy logic?